token-sdk/README.md

```diff
@@ -0,0 +1,411 @@
+
+Install tokengen. Tokengen is a tool to create the configuration file for the token chaincode (once, when deploying the chaincode). It generates the public parameters that the network participants will use to generate their proofs, and it specifies the public identities of the issuer, auditor and CA for signature validation.
+
+```bash
+go install github.com/hyperledger-labs/fabric-token-sdk/cmd/tokengen@v0.3.0
+```
+
+### Quick start
+
+The quickest way to get going is to run:
+
+```bash
+./scripts/up.sh
+```
+
+This generates the crypto material, starts Fabric, deploys the chaincode, and starts the token nodes.
+
+When you're done and want to delete everything:
+
+```bash
+./scripts/down.sh
+```
+
+#### Using the application
+
+The services are accessible on the following ports:
+
+| port | service                  |
+|------|--------------------------|
+| 8080 | API documentation (web)  |
+| 9000 | auditor                  |
+| 9100 | issuer                   |
+| 9200 | owner 1 (alice and bob)  |
+| 9300 | owner 2 (carlos and dan) |
+
+Besides that, the nodes communicate with each other via 9001, 9101, 9201 and 9301 respectively.
+
+Now let's issue and transfer some tokens! View the API documentation and try some actions at [http://localhost:8080](http://localhost:8080). Or, directly from the commandline:
+
+```bash
+curl -X POST http://localhost:9100/api/v1/issuer/issue -H 'Content-Type: application/json' -d '{
+    "amount": {"code": "EURX","value": 1000},
+    "counterparty": {"node": "owner1","account": "alice"},
+    "message": "hello world!"    
+}'
+
+curl -X GET http://localhost:9200/api/v1/owner/accounts
+
+curl -X POST http://localhost:9200/api/v1/owner/accounts/alice/transfer -H 'Content-Type: application/json' -d '{
+    "amount": {"code": "EURX","value": 100},
+    "counterparty": {"node": "owner1","account": "bob"},
+    "message": "hello bob!"    
+}'
+
+curl -X GET http://localhost:9200/api/v1/owner/accounts/bob/transactions
+curl -X GET http://localhost:9200/api/v1/owner/accounts/alice/transactions
+```
+
+Notice that the transaction overview uses the UTXO model (like bitcoin). The issuer created a new EURX token of 1000 and assigned its ownership to alice. When alice transfered 100 EURX to bob, she used the token of 1000 as **input** for her transaction. As **output**, she creates two new tokens:
+
+1. one for 100 EURX with bob as the owner
+2. one with _herself_ as the owner for the remaining 990 EURX.
+
+This way, each transaction can have multiple inputs and multiple outputs. Their sum should always be the same, and every new transfer has to be based on previously created outputs.
+
+#### Deep dive: what happens when doing a transfer?
+
+It may look simple from the outside, but there's a lot going on to securely and privately transfer tokens. Let's take an example of alice (on the Owner 1 node) transfering 100 TOK to dan (on the Owner 2 node).
```

Comment:
Isn't it a transfer of 100 EURX to bob? Although maybe you want to describe a node 2 scenario...

Reply:
Good point, I changed the curl commands to also transfer 100 TOK to dan to align the scenarios.